### PR TITLE
Fix missing column and variable reference before assignment

### PR DESCRIPTION
**Pull request by annajung**
- Add 'Number' column during unannotated data creation 
   - Fixes `KeyError` when running `run.py` due to missing 'Number' column in the annotated data
- Fix referencing variables before assignment issue
- Add input validation for model argument
- Update README with the correct path

### CODE REVIEW

**Comment by annajung** [comment]
@difince @enyinna1234 @pramodrj07 @tzstoyanov PTAL thanks!

**Comment by pramodrj07** [comment]
Are the changes in README.md anyway related to the other commits?

File: ml-conversational-analytic-tool/run.py [hunk 1]
```diff
@@ -16,20 +15,18 @@ def run(annotated_filename, dataset_filename, outcome, encoding_type, model_type
...
-    if encoding_type == 'role':
-        obs, res = data.getRoleMatrix(outcome, padding)
-    elif encoding_type == 'role-agnostic':
-        obs, res = data.getRoleAgnosticMatrix(outcome, padding)
-
     # Create models
-    if model_type == 'CNN':
-        model = BaseCNN()
-    elif model_type == 'LSTM':
+    if model_type == 'LSTM':
```
Comment (pramodrj07) [inline ml-conversational-analytic-tool/run.py:19]: Since i am still trying to understand the code base, Can i know the intention behind this change?

Comment (annajung) [reply ml-conversational-analytic-tool/run.py:19]: When a user passes a model type that is not 'CNN' or 'LSTM', it throws variable reference before assignment error. Therefore, I made CNN the default by removing `elif` to make sure the `model` is always initialized. I also added input validation to make sure that users can pass in only CNN and LSTM.

**Comment by annajung** [comment]
> Are the changes in README.md anyway related to the other commits?

@pramodrj07 no, it's a separate commit to update just the docs with correct path to run the files

**Comment by pramodrj07** [comment]
> 

Okay!! LGTM. 
Just as a rule of thumb. It would be good to demarcate(or have a separate PR), if a commit tends to deviate from a PR's primary intention.

File: ml-conversational-analytic-tool/run.py [hunk 2]
```diff
@@ -51,6 +51,10 @@ def run(annotated_filename, dataset_filename, outcome, encoding_type, model_type
     parser.add_argument('-pad', action='store_true', default=False, help='Pad total length of each pull')
 
     args = parser.parse_args()
+
+    if args.model != 'CNN' and args.model != 'LSTM':
+        raise Exception("Model must be either CNN or LSTM")
```
Comment (tzstoyanov) [inline ml-conversational-analytic-tool/run.py:56]: I would suggest to add supported model in the argument's help string also, something like that:
`parser.add_argument('model', help='Model type to use for training, supported CNN and LSTM')`

File: ml-conversational-analytic-tool/run.py (hunk 1 above)
Comment (tzstoyanov) [reply ml-conversational-analytic-tool/run.py:19]: I think that setting a model default value should be in the argument definition:
`parser.add_argument('model', default='CNN', help='Model type to use for training')`
It is more straightforward and the code is clean.

**Review by tzstoyanov — CHANGES_REQUESTED** [review]

File: ml-conversational-analytic-tool/run.py (hunk 1 above)
Comment (annajung) [reply ml-conversational-analytic-tool/run.py:19]: Sorry for the confusion, should have not used the word default. I don't want to set CNN to default. The user should always explicitly pass in what algorithm they want to use with the flags that are appropriate for that training. 

With the input validation, else statement will always be CNN.

File: ml-conversational-analytic-tool/run.py (hunk 2 above)
Comment (annajung) [reply ml-conversational-analytic-tool/run.py:56]: Done! Thanks

File: ml-conversational-analytic-tool/run.py (hunk 1 above)
Comment (tzstoyanov) [reply ml-conversational-analytic-tool/run.py:19]: a, ok - didn't get that logic. So, in that case may be it is better to extend the validation - to exit with an error if the model parameter is missing ?

**Review by difince — APPROVED** [review]
Taking into consideration Tzvetomir's notes, it LGTM

File: ml-conversational-analytic-tool/run.py (hunk 1 above)
Comment (tzstoyanov) [reply ml-conversational-analytic-tool/run.py:19]: I see, so in that case model must be either CNN or LSTM in run(). Thanks!

**Review by tzstoyanov — APPROVED** [review]

**Review by enyinna1234 — APPROVED** [review]

File: ml-conversational-analytic-tool/featureVector.py [hunk 3]
```diff
@@ -181,9 +182,11 @@ def pullStringConversation(self, export_filename="", export=True):
                                                                           comment_row["Body"])
             string_conversations.append(conversation.encode("ascii", "ignore").decode())
             pull_urls.append(row["URL"])
+            pull_numbers.append(row["Number"])
 
         # Export converation field dataset
```
Comment (enyinna1234) [inline ml-conversational-analytic-tool/featureVector.py:187]: Typo: conversation

Comment (annajung) [reply ml-conversational-analytic-tool/featureVector.py:187]: good catch, looks like there's a lot of the same typos in multiple places. I will leave that out of this PR and make a separate PR for clean up

**Pull request merged**